lib/Sema/CSSimplify.cpp

```diff
@@ -5703,6 +5703,9 @@ bool ConstraintSystem::repairFailures(
   }
 
   case ConstraintLocator::OptionalPayload: {
+    recordAnyTypeVarAsPotentialHole(lhs);
```

Comment:
The underlying issue here is that pattern variables are not marked "holes" but they should be and that would be addressed by @ahoppen's PR https://github.com/apple/swift/pull/60062/files#diff-5fffd7ce67aca13059738c34fbc714dcbaf3f6fe83daff7ca75836bc88304c20R2330-R2333

Reply:
Ah I see, it does make sense. So for this one, should we wait until it landed and add just the regression test?

Reply:
@ahoppen Would pull the test-case into his PR.

Reply:
I just confirmed that it’s fixed as part of https://github.com/apple/swift/pull/60062. I also added the test case in that PR.

Reply:
Cool, closing this then. Thanks @xedin!

Reply:
Any time!